types/test/async-component-test.ts

```diff
@@ -0,0 +1,61 @@
+import Vue, { AsyncComponent, Component } from "../index";
+
+const a: AsyncComponent = () => ({
+  component: new Promise<Component>((res, rej) => {
+    res({ template: "" });
+  }),
+});
```

Comment:
can you adapt the style to follow the existing one? eslint --fix should remove the semi colons and extra commas